src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -184,6 +183,71 @@ TxQ::FeeMetrics::scaleFeeLevel(OpenView const& view,
     return baseLevel;
 }
 
+namespace detail {
+
+static
+std::pair<bool, std::uint64_t>
+sumOfFirstSquares(std::size_t x)
+{
+    // sum(n = 1->x) : n * n = x(x + 1)(2x + 1) / 6
+
+    // If x is anywhere on the order of 2^^21, it's going
+    // to completely dominate the computation and is likely
+    // enough to overflow that we're just going to assume
+    // it does. If we have anywhere near 2^^21 transactions
+    // in a ledger, this is the least of our problems.
+    if (x >= (1 << 21))
+        return std::make_pair(false,
+            std::numeric_limits<std::uint64_t>::max());
+    return std::make_pair(true, (x * (x + 1) * (2 * x + 1)) / 6);
+}
+
+}
+
+std::pair<bool, std::uint64_t>
+TxQ::FeeMetrics::escalatedSeriesFeeLevel(OpenView const& view,
+    std::size_t extraCount, std::size_t seriesSize) const
+{
+    /* Transactions in the open ledger so far.
+        AKA Tranasactions that will be in the open ledger when
+        the first tx in the series is attempted.
+    */
+    auto const current = view.txCount() + extraCount;
+    /* Tranasactions that will be in the open ledger when
+        the last tx in the series is attempted.
+    */
+    auto const last = current + seriesSize - 1;
+
+    auto const params = [&]() {
+        std::lock_guard <std::mutex> sl(lock_);
+        return std::make_pair(txnsExpected_,
+            escalationMultiplier_);
+    }();
+    auto const& target = params.first;
+    auto const& multiplier = params.second;
+
+    assert(current > target);
+
+    /* Calculate (apologies for the terrible notation)
+        sum(n = current -> last) : multiplier * n * n / (target * target)
+        multiplier / (target * target) * (sum(n = current -> last) : n * n)
+        multiplier / (target * target) * ((sum(n = 1 -> last) : n * n) -
+            (sum(n = 1 -> current - 1) : n * n))
+    */
+    auto const sumNlast = detail::sumOfFirstSquares(last);
+    auto const sumNcurrent = detail::sumOfFirstSquares(current - 1);
+    if (!sumNlast.first || !sumNcurrent.first)
```

Comment:
And the irony of course is that I used that same justification to return that single item. Good catch.